x-pack/plugins/stack_connectors/common/gen_ai/schema.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+// Connector schema
+export const GenAiConfigSchema = schema.object({
+  apiProvider: schema.string(),
+  apiUrl: schema.string(),
+});
+
+export const GenAiSecretsSchema = schema.object({ apiKey: schema.string() });
+
+// Run action schema
+export const GenAiRunActionParamsSchema = schema.object({
+  body: schema.string(),
+});
+export const GenAiRunActionResponseSchema = schema.object({}, { unknowns: 'ignore' });
```

Comment:
I do not think we need this schema.

Reply:
Confused... I am using it, and you are telling me to use it: https://github.com/elastic/kibana/pull/157228/files#r1203961063

Reply:
In `x-pack/plugins/stack_connectors/server/connector_types/gen_ai/gen_ai.ts` on the `runApi` function the return type is `Promise<GenAiRunActionResponse>`. If you follow the path `GenAiRunActionResponse = TypeOf<typeof GenAiRunActionResponseSchema>;` where `GenAiRunActionResponseSchema = schema.object({}, { unknowns: 'ignore' });`. `GenAiRunActionResponseSchema` is in `x-pack/plugins/stack_connectors/common/gen_ai/schema.ts . But what you actually return is x-pack/plugins/stack_connectors/server/connector_types/gen_ai/api_schema.ts` defined in `x-pack/plugins/stack_connectors/server/connector_types/gen_ai/gen_ai.ts`. So the correct return type of runApi is `Promise<typeof GenAiBaseApiResponseSchema>`.